packages/@aws-cdk/aws-codedeploy/README.md

```diff
@@ -1,13 +1,55 @@
 ## The CDK Construct Library for AWS CodeDeploy
 
+### Applications
+
+To create a new CodeDeploy Application that deploys to EC2/on-premise instances:
+
+```ts
+import codedeploy = require('@aws-cdk/aws-codedeploy');
+
+const application = new codedeploy.ServerApplication(this, 'CodeDeployApplication', {
+    applicationName: 'MyApplication', // optional property
+});
+```
+
+To import an already existing Application:
+
+```ts
+const application = codedeploy.ServerApplicationRef.import(this, 'ExistingCodeDeployApplication', {
+    applicationName: new codedeploy.ApplicationName('MyExistingApplication'),
+});
+```
+
+### Deployment Groups
+
+To create a new CodeDeploy Deployment Group that deploys to EC2/on-premise instances:
+
+```ts
+const deploymentGroup = new codedeploy.ServerDeploymentGroup(this, 'CodeDeployDeploymentGroup', {
+    application,
+    deploymentGroupName: 'MyDeploymentGroup',
+});
+```
+
+All properties are optional - if you don't provide an Application,
+one will be automatically created.
+
+To import an already existing Deployment Group:
+
+```ts
+const deploymentGroup = codedeploy.ServerDeploymentGroupRef.import(this, 'ExistingCodeDeployDeploymentGroup', {
+    application,
+    deploymentGroupName: new codedeploy.DeploymentGroupName('MyExistingDeploymentGroup'),
```

Comment:
Given the name of those - I would remove the `Server` part of the class names (`ServerDeploymentGroup`, `ServerApplication`).

Reply:
We can't - we'll have `LambdaApplication` and `LambdaDeploymentGroup` soon, and in the future `ContainerApplication` and `ContainerDeploymentGroup`, and possibly others.